src/event/ngx_event_openssl_cache.c

```diff
@@ -175,12 +204,46 @@ ngx_ssl_cache_fetch(ngx_conf_t *cf, ngx_uint_t index, char **err,
     cn->id.len = id.len;
     cn->id.type = id.type;
     cn->type = type;
+    cn->value = NULL;
 
     ngx_cpystrn(cn->id.data, id.data, id.len + 1);
 
-    cn->value = type->create(&id, err, data);
+    if (id.type == NGX_SSL_CACHE_PATH) {
+
+        if (ngx_file_info(cn->id.data, &fi) != NGX_FILE_ERROR) {
+            cn->mtime = ngx_file_mtime(&fi);
+            cn->uniq = ngx_file_uniq(&fi);
+        }
+    }
+
+    /* try to use a reference from the old cycle */
+
+    old_cache = ngx_ssl_cache_get_old_conf(cf->cycle);
+
+    if (old_cache && old_cache->inherit) {
```

Comment:
Shouldn't we check the current `cache->inherit` here?

Reply:
There are two reasons why we should check rather a previous cycle configuration:
- logical
It is more consistent to state that configuration defines whether this configuration cycle should be prepared to inherit objects from *this* cycle to the next one.
Further, we can state that if the inheritance is not allowed from this cycle, we can optimize out such storage once parsing has finished (e.g., demolish rbtree, etc).
- technical
It is not possible to start using configuration while is in the middle of configuration parsing, because the merge phase did not yet happen.

Reply:
I agree with the technical reason, but not fully convinced in logical:
* It means that the server state is not defined by the current config. I haven't encountered any directives in the open-source code explicitly affecting the next reload.
  There are some scenarios around `listen`, but that's more of a limitation that an intentional behavior.
* Two reloads are required to flush the cache.

At the very least, we should consider renaming `inherit` -> `inheritable` to point that it's not an instruction to the config parser but rather a sign of the old cache being inheritable.
Then we could try to draft the directive doc with various pitfalls and see if it can be explained clearly to the users.

Reply:
> technical
It is not possible to start using configuration while is in the middle of configuration parsing, because the merge phase did not yet happen.

Indeed, well not without requiring the user put the directive at the top of the configuration / prior to parsing of any other configuration that will call ngx_ssl_cache_fetch - load_module being an example of a directive that is obviously order dependent like this.

>Two reloads are required to flush the cache.

This would be annoying.

Reply:
Changed to inheritable.

Reply:
> * Two reloads are required to flush the cache.

The directive isn't used to be altered frequently, so there is no issue. 

The value should either "on" (a reasonable default), or "off" to give a workaround in case something happens to not work as expected.